File: solutions/cpp/nth-prime/1/nth_prime.cpp
#include "nth_prime.h"

#include <stdexcept>	
#include <vector>

namespace nth_prime {

    unsigned int nth(unsigned int num) {
        if (num < 1) {
            throw std::domain_error("");
        }
       std::vector<unsigned int> primes{2};
       for(unsigned int current{3}; primes.size() < num; current += 2) {
           bool is_prime{true};
           for(auto prime : primes) {
               if (current % prime == 0) {
                   is_prime = false;
                   break;
               }
           }
           if(is_prime) {
               primes.emplace_back(current);
           }
           
       }
        
        return primes.back();
    }

}  // namespace nth_prime

File: solutions/csharp/nth-prime/1/NthPrime.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class NthPrime
{
    public static int Prime(int nth)
    {
        if (nth == 0)
        {
            throw new ArgumentOutOfRangeException();
        }
        int num = 1;
        int count = 0;
        int i;

        while (count < nth)
        {
            num++;
            for (i = 2; i <= num; i++)
            {
                if (num % i == 0)
                {
                    break;
                }
            }
            if (i == num)
            {
                count = count + 1;
            }
        }
        return num;
    }
}